feat(flags): document custom featureFlagsIntegration and refactor all JS integration docs (#12151)
getsentry/sentry-docs: docs/platforms/javascript/common/configuration/integrations/featureFlags.mdx

---
title: Feature Flags
description: "Learn how to attach custom feature flag data to Sentry error events."
notSupported:
  - javascript.aws-lambda
  - javascript.azure-functions
  - javascript.bun
  - javascript.capacitor
  - javascript.cloudflare
  - javascript.connect
  - javascript.cordova
  - javascript.deno
  - javascript.electron
  - javascript.express
  - javascript.fastify
  - javascript.gcp-functions
  - javascript.hapi
  - javascript.koa
  - javascript.nestjs
  - javascript.nodejs
  - javascript.wasm
---

<PlatformContent includePath="feature-flags/prerelease-alert" />

<Alert level="info">

This integration only works inside a browser environment.

</Alert>

> The Feature Flags integration allows you to manually track feature flag evaluations through an API. These evaluations are held in memory, and in the event an error occurs, sent to Sentry for review and analysis.

Reviewer: Maybe simpler and avoid `event` which is kind of an overloaded term for us.

```suggestion
The Feature Flags integration allows you to manually track feature flag evaluations through an API. These evaluations are held in memory and sent to Sentry when an error occurs.
```

**At the moment, we only support boolean flag evaluations.**

_Import names: `Sentry.featureFlagsIntegration` and `type Sentry.FeatureFlagsIntegration`_

## Install

Install your platform's Sentry SDK from npm.

## Configure

```JavaScript

> import * as Sentry from '@sentry/<your browser platform, e.g. react>';

Reviewer: 😬 you'll want to have install code for each platform unfortunately and then plop something like this in here:

```
<PlatformContent includePath="feature-flags/install" />
```

Author: oof we need this for other FF integrations too. Maybe there's an existing platform content for it 👀
Or maybe we can exclude this line since we already have the "Import names"?

Sentry.init({
  dsn: "___PUBLIC_DSN___",
  integrations: [Sentry.featureFlagsIntegration()]
});
```

## Verify

The integration is tested by calling the `addFeatureFlag` method before capturing an exception.

```JavaScript
import * as Sentry from '@sentry/browser';

const flagsIntegration = Sentry.getClient()?.getIntegrationByName<Sentry.FeatureFlagsIntegration>('FeatureFlags');
if (flagsIntegration) {
  flagsIntegration.addFeatureFlag('hello', false);
} else {
  // check your configure step
}

> Sentry.captureException(Exception('broke'));

Reviewer: if you're going to merge this... this is not valid

```suggestion
Sentry.captureException(new Error('broke'));
```

Author: Oof ok I'll change in other providers too

```

> Visit the Sentry website and confirm that your error event has recorded the feature flag "hello" and its value "false".

Reviewer: Not sure if we need a "Verify" section... maybe just "Usage"?

Author: We use Verify for all the LD and OF docs. Not sure where the convention started, think some other integrations do it?

Reviewer: OK, I would poke around and see what other integrations do
